accounts/views.py

```diff
@@ -212,12 +205,34 @@ def get(self, request, *args, **kwargs):
             "unpaid_amount": unpaid_amount,
             "unpaid_amount_text": cents_to_dollars(unpaid_amount),
         }
+        # Pass referral link
+        referral_link = get_refferal_link(request.user.pk)
+        context["reflink"] = request.build_absolute_uri(referral_link)
+
         # Pass first_signup variable to tell Google Analytics
         # if this signup should be counted as a conversion.
         if hasattr(self.request, "session"):
             first_signup = self.request.session.pop("first_signup", False)
             if first_signup:
                 context["first_signup"] = True
+                # Add refferal to reffered account
+                try:
+                    referral_hit = ReferralHit.objects.filter(
+                        hit_user_id=request.user.pk
+                    ).latest("created")
+                except ReferralHit.DoesNotExist:
+                    referral_hit = None
+                if referral_hit is not None:
+                    try:
+                        referred_user = referral_hit.referral_link.user
+                        UserProfileReferralHit.objects.create(
+                            referral_hit=referral_hit,
+                            user_profile=referred_user,
+                        )
+                        referral_hit.confirmed = timezone.now()
+                        referral_hit.save()
+                    except UserProfileReferralHit.DoesNotExist:
```

Comment:
@autn the try...except block should wrap as little code as possible, so it shouldn't wrap the save method, for instance.